src/test/java/com/ksm/wstbackoffice/controller/WarehouseControllerIT.java

```diff
@@ -0,0 +1,135 @@
+})
+public class WarehouseControllerIT extends BaseControllerIT {
+    @Autowired
+    private WarehouseService warehouseService;
+    @Autowired
+    private AddressService addressService;
+    @Autowired
+    private WarehouseMapper warehouseMapper;
+    @Autowired
+    private AddressMapper addressMapper;
+
+
+    public static Stream<Arguments> findByIdTestArguments() {
+        return Stream.of(
+                Arguments.of(-1L, 200),
+                Arguments.of(-100L, 404),
+                Arguments.of(0L, 404),
+                Arguments.of(Long.MAX_VALUE, 404)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("findByIdTestArguments")
+    public void findByIdTest(Long id, int expectedStatus) {
+        given().
+            pathParam("id", id).
+        when().
+            get("warehouses/{id}").
+        then().
+            statusCode(expectedStatus);
+    }
+
+    @Test
+    public void findAllTest() {
+        when().
+            get("warehouses").
+        then().
+            statusCode(200).
+        body("size()", CoreMatchers.equalTo(4));
+    }
+
+    @Test
+    public void createTest() {
+        AddressEntity address = new AddressEntity();
+        address.setApartmentNumber("9");
+        address.setCity("Kiev");
+        address.setDistrict("Podil district");
+        address.setHouseNumber("14a");
+        address.setRegion("Podil");
+        address.setStreet("Sagaidachnogo");
+        address.setZipcode("01032");
+        address.setDescription("description address");
+
+        WarehouseDto warehouseDto = new WarehouseDto();
+        warehouseDto.setName("warehause_5");
+        warehouseDto.setWarehouseType(WarehouseType.RETAIL);
+        warehouseDto.setPriceType(PriceType.RETAIL);
+        warehouseDto.setDescription("description_5");
+        warehouseDto.setAddress(address);
+
+        given().
+            contentType(ContentType.JSON).
+            body(warehouseDto).
+        when().
+            post("warehouses").
+        then().
+            statusCode(201).
+            body("id", notNullValue()).
+            body("name", equalTo("warehause_5")).
+            body("warehouseType", equalTo("RETAIL")).
+            body("priceType", equalTo("RETAIL"));
+    }
+
+    @Test
+    public void updateTest() {
+        int warehouse_id = -1;
+        int address_id = -4;
+
+        AddressDto addressDto = addressService.findById(address_id);
+
+        WarehouseDto warehouseDto = warehouseService.findById(warehouse_id);
+        WarehouseEntity warehouseEntity = warehouseMapper.toEntity(warehouseDto);
+        warehouseEntity.setName("warehouse_update_1");
+        warehouseEntity.setPriceType(PriceType.WHOLESALE);
+        warehouseEntity.setWarehouseType(WarehouseType.WHOLESALE);
+        warehouseEntity.setDescription("warehouse_update_description_1");
+        warehouseEntity.setAddress(addressMapper.toEntity(addressDto));
```

Comment:
Do you think you really need warehouseEntity here?

Reply:
fixed